driver.ts

```diff
@@ -36,13 +36,33 @@ export async function reverse(
   await sendText(device, "\x1DB" + enabled);
 }
 
+/**
+ * Select an endpoint from the currently selected device configuration by name
+ * 
+ * @param name usually 'out' or 'in'
+ * @param device the connected USBDevice
+ */
+function selectEndpoint(name: string, device: USBDevice) {
+  const endpoint = device.configuration
+    .interfaces[0]
+    .alternate
+    .endpoints.filter(ep => ep.direction == name)
```

Comment:
prefer `.find()` instead of filter + shift



--


filter + shift is cool though, I have never seen this approach before :)

Reply:
This is awkward. I never knew there was a find method, haha. I guess that filter shift comes from functional stuff I did some time ago.